pkg/controller/hibernation/aws_actuator.go

```diff
@@ -149,7 +275,32 @@ func getAWSClient(cd *hivev1.ClusterDeployment, c client.Client, logger log.Fiel
 	return awsclient.New(c, options)
 }
 
-func getClusterInstanceIDs(cd *hivev1.ClusterDeployment, c awsclient.Client, states sets.String, logger log.FieldLogger) ([]*string, error) {
+// filterOutSpotInstances removes the spot instances from the list and returns it. It
+// also returned the spot instances that were filtered out in a separate list.
+func filterOutSpotInstances(instances []*ec2.Instance) ([]*ec2.Instance, []*ec2.Instance) {
+	var spotInstances []*ec2.Instance
+	n := 0
+	for _, i := range instances {
+		if aws.StringValue(i.InstanceLifecycle) == "spot" {
+			spotInstances = append(spotInstances, i)
+			continue
+		}
+		instances[n] = i
+		n++
+	}
+	instances = instances[:n]
+	return instances, spotInstances
```

Comment:
this is a common technique of filtering a slice. https://github.com/golang/go/wiki/SliceTricks#filter-in-place
IMO, the name of the function makes it clear that it will remove certain objects from the slice.